swarm_nl/src/core/prelude.rs

```diff
@@ -247,214 +256,136 @@ impl StreamResponseBuffer {
-		// Default implementation
-	}
-
+	KademliaStartProvidingSuccess { key: Vec<u8> },
 	/// Event that announces the failure of a node to become a provider of a record in the DHT.
-	fn kademlia_start_providing_error(&mut self) {
-		// Default implementation
-	}
-
+	KademliaStartProvidingError,
 	/// Event that announces the arrival of an RPC message.
-	fn rpc_incoming_message_handled(&mut self, data: Vec<Vec<u8>>) -> Vec<Vec<u8>>;
-
+	RpcIncomingMessageHandled { data: RpcData },
 	/// Event that announces that a peer has just left a network.
-	fn gossipsub_unsubscribe_message_recieved(&mut self, _peer_id: PeerId, _topic: String) {
-		// Default implementation
-	}
-
+	GossipsubUnsubscribeMessageReceived { peer_id: PeerId, topic: String },
 	/// Event that announces that a peer has just joined a network.
-	fn gossipsub_subscribe_message_recieved(&mut self, _peer_id: PeerId, _topic: String) {
-		// Default implementation
-	}
-
+	GossipsubSubscribeMessageReceived { peer_id: PeerId, topic: String },
 	/// Event that announces the arrival of a gossip message.
-	fn gossipsub_incoming_message_handled(&mut self, _source: PeerId, _data: Vec<String>);
-
+	GossipsubIncomingMessageHandled { source: PeerId, data: Vec<String> },
 	/// Event that announces the beginning of the filtering and authentication of the incoming
-	/// gossip message. It returns a boolean to specify whether the massage should be dropped or
+	/// gossip message. It returns a boolean to specify whether the message should be dropped or
 	/// should reach the application. All incoming messages are allowed in by default.
-	fn gossipsub_incoming_message_filtered(
-		&mut self,
-		_propagation_source: PeerId,
-		_message_id: MessageId,
-		_source: Option<PeerId>,
-		_topic: String,
-		_data: Vec<String>,
-	) -> bool {
-		true
-	}
+	GossipsubIncomingMessageFiltered {
+		propagation_source: PeerId,
+		message_id: MessageId,
+		source: Option<PeerId>,
+		topic: String,
+		data: Vec<String>,
+	},
 }
 
-/// Default network event handler.
-#[derive(Clone)]
-pub struct DefaultHandler;
-/// Implement [`EventHandler`] for [`DefaultHandler`]
-impl EventHandler for DefaultHandler {
-	/// Echo the message back to the sender.
-	fn rpc_incoming_message_handled(&mut self, data: Vec<Vec<u8>>) -> Vec<Vec<u8>> {
-		data
-	}
-
-	/// Echo the incoming gossip message to the console.
-	fn gossipsub_incoming_message_handled(&mut self, _source: PeerId, _data: Vec<String>) {
-		// Default implementation
-	}
+/// The struct that contains incoming information about a peer returned by the `Identify` protocol.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct IdentifyInfo {
+	/// The public key of the remote peer
+	pub public_key: PublicKey,
+	/// The address the remote peer is listening on
+	pub listen_addrs: Vec<Multiaddr>,
+	/// The protocols supported by the remote peer
+	pub protocols: Vec<StreamProtocol>,
+	/// The address we are listened on, observed by the remote peer
+	pub observed_addr: Multiaddr,
```

Comment:
```suggestion
/// The struct that contains incoming information about a peer returned by the `Identify` protocol.
#[derive(Debug, Clone, Eq, PartialEq, Hash)]
pub struct IdentifyInfo {
	/// The public key of the remote peer.
	pub public_key: PublicKey,
	/// The address the remote peer is listening on.
	pub listen_addrs: Vec<Multiaddr>,
	/// The protocols supported by the remote peer.
	pub protocols: Vec<StreamProtocol>,
	/// The address we are listened on, observed by the remote peer.
	pub observed_addr: Multiaddr,
```

How can we say "/// The address we are listened on, observed by the remote peer." more clearly? Maybe:
" Our address that the remote peer is listening on" ? 🤔

Reply:
No. The address that the remote peer thinks we are listening on.